recipes/armadillo/all/conanfile.py

```diff
@@ -0,0 +1,414 @@
+            "use_lapack",
+        ],
+    }
+    exports_sources = "CMakeLists.txt"
+    generators = (
+        "cmake",
+        "cmake_find_package",
+    )
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        if self.settings.os == "Macos":
+            # Macos will default to Accelerate framework
+            self.options.use_blas = "framework_accelerate"
+            self.options.use_lapack = "framework_accelerate"
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+        # Use default MKL path or throw error if no default path exists
+        if self.options.use_blas == "system_mkl" and not self.options.mkl_library_path:
+            if self.settings.os == "Linux":
+                self.options.mkl_library_path = "/opt/intel/mkl/lib/intel64"
+            elif self.settings.os == "Windows":
+                self.options.mkl_library_path = (
+                    "C:/IntelSWTools/compilers_and_libraries/windows/mkl/lib/intel64"
+                )
+            else:
+                raise ConanInvalidConfiguration(
+                    "A default mkl_library_path value for your operating system is not available. Please specify a value for mkl_library_path"
+                )
+
+    def validate(self):
+        tools.check_min_cppstd(self, "11")
+
+        if self.settings.os != "Macos" and (
+            self.options.use_blas == "framework_accelerate"
+            or self.options.use_lapack == "framework_accelerate"
+        ):
+            raise ConanInvalidConfiguration(
+                "framework_accelerate can only be used on Macos"
+            )
+
+        for value, options in self.co_dependencies.items():
+            options_without_value = [
+                x for x in options if getattr(self.options, x) != value
+            ]
+            if options_without_value and (len(options) != len(options_without_value)):
+                raise ConanInvalidConfiguration(
+                    "Options {} must all be set to '{}' to use this feature. To fix this, set option {} to '{}'.".format(
+                        ", ".join(options),
+                        value,
+                        ", ".join(options_without_value),
+                        value,
+                    )
+                )
+
+        if (
+            self.options.use_lapack == "openblas"
+            and self.options.use_blas != "openblas"
+        ):
+            raise ConanInvalidConfiguration(
+                "OpenBLAS can only provide LAPACK functionality when also providing BLAS functionality. Set use_blas=openblas and try again."
+            )
+
+    def requirements(self):
+        # Optional requirements
+        openblas = "openblas/0.3.15"
```

Comment:
Don't need to declare here, call directly from the method, you are not re-using this reference, only allocating more memory.